server/model/monitor.js

```diff
@@ -809,6 +809,24 @@ class Monitor extends BeanModel {
                 bean.important = true;
 
                 if (Monitor.isImportantForNotification(isFirstBeat, previousBeat?.status, bean.status)) {
+                    // Change the notification msg if status becomes UP from DOWN
+                    // Adding msg for the downtime
+                    if (bean.status === UP) {
+                        const downAfterPreviousUpBeat = await R.findOne("heartbeat", " monitor_id = ? AND status = ? AND time > ( SELECT time from heartbeat WHERE monitor_id = ? AND status = ? ORDER BY time DESC ) ORDER BY time ASC", [
+                            this.id,
+                            0,
+                            this.id,
+                            1
+                        ]);
+                        if (downAfterPreviousUpBeat) {
+                            const mostRecentDownBeat = await R.findOne("heartbeat", " monitor_id = ? AND status = ? ORDER BY time DESC", [
+                                this.id,
+                                0
+                            ]);
+                            const downTime = dayjs(mostRecentDownBeat?.time).diff(dayjs(downAfterPreviousUpBeat?.time), "minutes", true);
+                            bean.msg = `${bean.msg} <Down for ${Math.floor(downTime)} mintue(s) ${(downTime % 1).toFixed(2).replace(/^0\./, "")} second(s)>`;
```

Comment:
Adding a message to the `msg` is something which might not be ideal.
This prevents notification providers like `discord`/`slack` to including them in the fancy embeds.

This is something that Luis will have to give his opinion on (both paths are workable imo)

Reply:
I think a better solution would be to add a parameter `downDuration` to send(), which is the duration of downtime in seconds, then individual notification providers can format the message as needed. This would provide better customization, and be non-breaking for the existing notifications.

Reply:
@chakflying I guess we will send the same msg to every notification provider. For instance, we will send how many minutes or hours this service goes down in the same manner to every provider. In refactored code, I have put the syntax like (mm: ss) format.

Reply:
If you check provider `discord.js`, it doesn't even use the `msg` parameter for up/down notifications. To be clear, I'm not saying this PR needs to update all the notification providers. You can update the ones that you are using. I'm just saying this is not a good solution considering the current situation.

Reply:
> If you check provider discord.js, it doesn't even use the msg parameter for up/down notifications.

Oh, I now get your point. Thanks for letting me know. I will add one more parameter in the send() function which will have downDuration value in seconds and then all notification providers can use it the way they want.

Reply:
I think this change should be added to the `heartbeatJSON` instead. 
What is the rationale behind adding another parameter?

If adding it to `heartbeatJSON` is not an option / adding a parameter this is better, please add this parameter to every function invocation to make this obvious to the authors of the notificationproviders)

Reply:
I have added downTime to hearbeatJSON. 
![code1](https://github.com/louislam/uptime-kuma/assets/56472269/f7bacba3-592e-440c-903b-30bdf62722d8)


But in the "sendNotification" function I have kept the downTime parameter and not added another property to the "bean" parameter as eventually we are storing the data of "bean" into the database and adding a property to it will mean either we have to add coulmn in table which is of no use or we have to delete that added (downTime) property from the "bean" object.

![code](https://github.com/louislam/uptime-kuma/assets/56472269/09ecf323-1232-47f2-854a-f313f40b5076)